internal/controller/config/checkout_params.go

```diff
@@ -34,7 +41,20 @@ func (co *CheckoutParams) ApplyToAgentStart(podSpec *corev1.PodSpec, ctr *corev1
 	// TODO: Agent start doesn't know about submodule clone config, but
 	// agent bootstrap does...
 	appendCommaSepToEnv(ctr, "BUILDKITE_GIT_SUBMODULE_CLONE_CONFIG", co.SubmoduleCloneConfig)
-	co.GitMirrors.ApplyTo(podSpec, ctr)
+
+	co.GitMirrors.ApplyToAgentStart(ctr)
+}
+
+// Any k8s related config things need to be passed to checkout container directly.
+// "kubernetes-bootstrap" won't work for those for obvious reason: they are passed k8s pod lifecycle.
+//
+// NOTE: despite this is called ApplyToCheckout, it mutate not only the container spec but also the pod spec.
```

Comment:
```suggestion
// NOTE: despite this is called ApplyToCheckout, it mutate not only the container spec but also the pod spec.
```

Maybe we should do the `GitMirrors.ApplyToPod` call from the scheduler instead